### ISSUE TITLE
ERR_BadEmbeddedStmt "Embedded statement cannot be a declaration or labeled statement" is not reported for a [let] statement 

### ISSUE THREAD

**Issue by AlekseyTs** [dotnet/roslyn#9029]
```
    void Test3()
    {
        if (true)
            var x18 = 11;

        if (true)
            let y18 = 11;
    }
```

Observed:

```
    // (11,13): error CS1023: Embedded statement cannot be a declaration or labeled statement
    //             var x18 = 11;
    Diagnostic(ErrorCode.ERR_BadEmbeddedStmt, "var x18 = 11;").WithLocation(11, 13)
```

Expected:
ERR_BadEmbeddedStmt should also be reported for the let statement.

**Comment by gafter** [comment]
The let statement is not part of the latest patterns design.
